modules/dnn_superres/src/dnn_superres.cpp

```diff
@@ -0,0 +1,385 @@
+                    Mat preproc_img;
+                    preprocess_YCrCb(img, preproc_img);
+
+                    //Split the image: only the Y channel is used for inference
+                    Mat ycbcr_channels[3];
+                    split(preproc_img, ycbcr_channels);
+
+                    Mat Y = ycbcr_channels[0];
+
+                    //Create blob from image so it has size 1,1,Width,Height
+                    cv::Mat blob;
+                    dnn::blobFromImage(Y, blob, 1.0);
+
+                    //Get the HR output
+                    this->net.setInput(blob);
+
+                    Mat blob_output = this->net.forward();
+
+                    //Convert from blob
+                    std::vector <Mat> model_outs;
+                    dnn::imagesFromBlob(blob_output, model_outs);
+                    Mat out_img = model_outs[0];
+
+                    //Reconstruct: upscale the Cr and Cb space and merge the three layer
+                    reconstruct_YCrCb(out_img, preproc_img, img_new, this->sc);
+                }
+                else if( this->alg == "edsr" )
+                {
+                    //BGR mean of the Div2K dataset
+                    Scalar mean =  Scalar(103.1545782, 111.561547, 114.35629928);
+
+                    //Convert to float
+                    Mat float_img;
+                    img.convertTo(float_img, CV_32F, 1.0);
+
+                    //Create blob from image so it has size [1,3,Width,Height] and subtract dataset mean
+                    cv::Mat blob;
+                    dnn::blobFromImage(float_img, blob, 1.0, Size(), mean);
+
+                    //Get the HR output
+                    this->net.setInput(blob);
+                    Mat blob_output = this->net.forward();
+
+                    //Convert from blob
+                    std::vector <Mat> model_outs;
+                    dnn::imagesFromBlob(blob_output, model_outs);
+                    img_new = model_outs[0];
+
+                    //Post-process: add mean.
+                    img_new = img_new + Scalar(103.1545782, 111.561547, 114.35629928);
+
+                    img_new.convertTo(img_new, CV_8U);
+                }
+                else
+                {
+                    //
+                }
+            }
+            else
+            {
+                std::cout << "Model not specified. Please set model via setModel(). \n";
+            }
+        }
+
+        void DnnSuperResImpl::upsampleMultioutput(Mat img, std::vector<Mat> &imgs_new, std::vector<int> scale_factors, std::vector<String> node_names)
+        {
+            CV_Assert(scale_factors.size() == node_names.size());
+            CV_Assert(!scale_factors.empty());
+            CV_Assert(!node_names.empty());
+
+            if ( this->alg != "lapsrn" )
+            {
+                std::cout << "Only LapSRN support multiscale upsampling for now!" << std::endl;
+                return;
```

Comment:
Please use `CV_Error()` instead of stdout.

Reply:
I replaced them with CV_Error.